### PR TITLE
Refactor FXIOS-10145 - Update how we are tearing down our tests

### PR DESCRIPTION
**Pull request by cavaevinci**
In teardown methods in test,put super.tearDown to be executed last during teardown

## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-10145)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/22232)

## :bulb: Description
In teardown methods in test,put super.tearDown to be executed last during teardown

## :pencil: Checklist
You have to check all boxes before merging
- [✏️ ] Filled in the above information (tickets numbers and description of your work)
- [✏️ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [✏️ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: BrowserKit/Package.resolved [hunk 1]
```diff
@@ -41,8 +41,17 @@
...
+        "version" : "8.26.0"
+      }
+    },
+    {
+      "identity" : "swiftdraw",
+      "kind" : "remoteSourceControl",
+      "location" : "https://github.com/swhitty/SwiftDraw",
+      "state" : {
+        "revision" : "a5c680f07b33f4cc9a451491b417b8119de23c6e",
+        "version" : "0.17.0"
```
Comment (cyndichin) [inline BrowserKit/Package.resolved:54]: lets keep the same resolved file and remove these changes

**Review by cyndichin — COMMENTED** [review]
thank you @cavaevinci for going through all the test files and making the changes 🥳 !! I had two small comments, but everything else looks great. 

@clarmso I'll leave you to confirm if there's any issues with the UI test, but it seems it passes bitrise.

File: BrowserKit/Package.resolved (hunk 1 above)
Comment (cavaevinci) [reply BrowserKit/Package.resolved]: @cyndichin  removed BrowserKit/Package.resolved file from PR

File: firefox-ios/firefox-ios-tests/Tests/ClientTests/OnboardingTests/OnboardingTelemetryDelegationTests.swift [hunk 2]
```diff
@@ -20,8 +20,8 @@ class OnboardingTelemetryDelegationTests: XCTestCase {
     }
 
     override func tearDown() {
-        nimbusUtility = nil
```
[the remark this answers is not in the document]
Comment (cavaevinci) [reply firefox-ios/firefox-ios-tests/Tests/ClientTests/OnboardingTests/OnboardingTelemetryDelegationTests.swift]: @cyndichin  my bad,I missed that one,we should also put tearDown after everyting is cleared.Fixed now.

File: BrowserKit/Package.resolved (hunk 1 above)
Comment (cyndichin) [reply BrowserKit/Package.resolved]: it seems the file was deleted, could we remove any changes from being committed related to that file instead? Basically, avoid having it show up in the commit changes.

File: firefox-ios/firefox-ios-tests/Tests/ClientTests/OnboardingTests/OnboardingTelemetryDelegationTests.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (cyndichin) [reply firefox-ios/firefox-ios-tests/Tests/ClientTests/OnboardingTests/OnboardingTelemetryDelegationTests.swift]: np, there were a lot of files to go through 😄 thank you for addressing that! looks good now :)

File: BrowserKit/Package.resolved (hunk 1 above)
Comment (cavaevinci) [reply BrowserKit/Package.resolved]: @cyndichin reverted to initial Package.resolved(hopefully).

**Review by cyndichin — APPROVED** [review]
thanks @cavaevinci, looks great now! appreciate the work!

we will also need an approval from the automation team cc: @clarmso

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/DragAndDropTests.swift [hunk 3]
```diff
@@ -21,10 +21,6 @@ let websiteWithSearchField = "https://developer.mozilla.org/en-US/"
 
 class DragAndDropTests: BaseTestCase {
 //  Disable test suite since in theory it does not make sense with Chron tabs implementation
-    override func tearDown() {
-        XCUIDevice.shared.orientation = UIDeviceOrientation.portrait
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/XCUITests/DragAndDropTests.swift:25]: We need to set the device to portrait again or the next tests would be run in landscape.

**Review by clarmso — CHANGES_REQUESTED** [review]

**Comment by clarmso** [comment]
> @clarmso I'll leave you to confirm if there's any issues with the UI test, but it seems it passes bitrise.

I can review the `XCUITests`.

File: firefox-ios/firefox-ios-tests/Tests/XCUITests/DragAndDropTests.swift (hunk 3 above)
Comment (cyndichin) [reply firefox-ios/firefox-ios-tests/Tests/XCUITests/DragAndDropTests.swift:25]: @cavaevinci wanted to follow up on this PR, seems like there's one outstanding comment from @clarmso

Comment (cavaevinci) [reply firefox-ios/firefox-ios-tests/Tests/XCUITests/DragAndDropTests.swift:25]: @cyndichin ,missed this one,sorry.I will resolve this in next 24 hours.

Comment (cyndichin) [reply firefox-ios/firefox-ios-tests/Tests/XCUITests/DragAndDropTests.swift:25]: no worries! thank you!

**Comment by cyndichin** [comment]
@cavaevinci unfortunately it seems there are some conflicts now, let me know if you are still interested in working on this PR; otherwise this PR will go stale and you have done great work!

**Comment by cyndichin** [comment]
@cavaevinci wanted to let you know that I will be away on PTO, I will tag @FilippoZazzeroni here to follow up on this PR while I am out. I will be back online on 12/16. Thank you!

**Pull request closed without merging**